awscli/autoprompt/widgets.py

```diff
@@ -0,0 +1,208 @@
+            help_doc_buffer, title='Doc panel help')
+        help_input_buffer = self.create_help_buffer(
+            'help_input', self.input_buffer_key_binding_text()
+        )
+        self.help_input_window = self.create_help_window(
+            help_input_buffer, title='Prompt panel help')
+        self.container = HSplit([
+            ConditionalContainer(
+                self.help_doc_window,
+                help_section_visible & (doc_window_has_focus
+                                        | search_input_has_focus)),
+            ConditionalContainer(
+                self.help_input_window,
+                help_section_visible & input_buffer_has_focus)
+        ])
+
+    def input_buffer_key_binding_text(self):
+        return (
+            f'{self.style}[ENTER]</style> Autocomplete '
+            f'Choice/Execute Command{self.eol}'
+            f'{self.style}[F1]</style> Hide/Show Help{self.eol}'
+            f'{self.style}[F2]</style> Focus on Docs{self.eol}'
+            f'{self.style}[F3]</style> Hide/Show Docs{self.eol}'
+            f'{self.style}[F4]</style> One/Multi column prompt{self.eol}'
+            f'{self.style}[CONTROL+R]</style> On/Off bck-i-search'
+        )
+
+    def doc_window_key_binding_text(self):
+        return (
+            f'{self.style}[/]</style> Search Forward{self.eol}'
+            f'{self.style}[?]</style> Search Backward{self.eol}'
+            f'{self.style}[n]</style> Find Next Match{self.eol}'
+            f'{self.style}[N]</style> Find Previous Match{self.eol}'
+            f'{self.style}[w]</style> Go Up a Page{self.eol}'
+            f'{self.style}[z]</style> Go Down a Page{self.eol}'
+            f'{self.style}[j]</style> Go Up a Line{self.eol}'
+            f'{self.style}[k]</style> Go Down a Line{self.eol}'
+            f'{self.style}[g]</style> Go to Top{self.eol}'
+            f'{self.style}[G]</style> Go to Bottom{self.eol}'
+            f'{self.style}[F1]</style> Hide/Show Help{self.eol}'
+            f'{self.style}[F2] or [q]</style> Focus on Input'
+        )
+
+    def create_help_buffer(self, name, help_text):
+        document = Document(text=help_text, cursor_position=0)
+        help_buffer = Buffer(name=name, read_only=True)
+        help_buffer.set_document(document, bypass_readonly=True)
+        return help_buffer
+
+    def create_help_window(self, help_buffer, title):
+        return Frame(
+            Window(
+                content=BufferControl(
+                    buffer=help_buffer,
+                    input_processors=[FormatTextProcessor()]
+                ),
+                width=Dimension(
+                    max=self.DIMENSIONS['help_window_width_max'],
+                    preferred=self.DIMENSIONS['help_window_width_pref']
+                ),
+                wrap_lines=True,
+            ),
+            title=title
+        )
+
+    def __pt_container__(self):
+        return self.container
+
+
+class ToolbarWidget:
```

Comment:
If we go with the feedback that I suggested with the `HelpPanelWidget`, I think we should do the same with the `ToolbarWidget` as well.